### PR TITLE
Add new regions and new instance types

### PR DESCRIPTION
**Pull request by amitaekbote**
## High Level Description

Change instance type and add new region support

## Related Issues

https://jira.mesosphere.com/browse/DCOS_OSS-12

## Checklist for all PR's

  - [x] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [x] Test Results: [link to CI job test results for component]
  - [x] Code Coverage (if available): [link to code coverage report]
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and Review Process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using github's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

File: gen/aws/templates/cloudformation.json [hunk 1]
```diff
@@ -317,7 +317,7 @@
         "SourceDestCheck" : "false",
         "KeyName" : { "Ref" : "KeyName" },
         "ImageId" : { "Fn::FindInMap" : [ "NATAmi", { "Ref" : "AWS::Region" }, "default" ] },
-        "InstanceType" : "m3.medium",
+        "InstanceType" : "m4.large",
```
Comment (BenWhitehead) [inline gen/aws/templates/cloudformation.json]: can you change this back to `m3.medium` I think this is what is breaking creation right now.

**Comment by amitaekbote** [comment]
@spahl @BenWhitehead @mellenburg defaulting to c4.xlarge instead of m3. 

Any additional test coverage apart from the defaults? ( I am already testing the new regions )

**Comment by ssk2** [comment]
Looks good to me and the @dcos/velocity-team!

**Comment by orsenthil** [comment]
@amitaekbote , the PR template needs an update to reflect the issue/state.

**Comment by lloesche** [comment]
@amitaekbote here's the complete list of regions and AMIs I build yesterday:

CentOS 7.3.1611 with Docker 1.13
```
ap-northeast-1:ami-d93d69be
ap-northeast-2:ami-542efe3a
ap-south-1:ami-6b0b7b04
ap-southeast-1:ami-68da6a0b
ap-southeast-2:ami-21fefd42
ca-central-1:ami-266ed342
eu-central-1:ami-e80bde87
eu-west-1:ami-68dcf20e
eu-west-2:ami-22cedb46
sa-east-1:ami-5c9cfa30
us-east-1:ami-b4f128a2
us-east-2:ami-ff82a79a
us-west-1:ami-d4b3edb4
us-west-2:ami-b863e1d8
us-gov-west-1:ami-b91ba1d8
```


And latest CoreOS 1235.12.0 from https://stable.release.core-os.net/amd64-usr/1235.12.0/coreos_production_ami_all.txt
```
ap-northeast-1=ami-86f1b9e1
ap-northeast-2=ami-0c15c562
ap-south-1=ami-4530402a
ap-southeast-1=ami-27cc7d44
ap-southeast-2=ami-5baeae38
ca-central-1=ami-a97bc6cd
eu-central-1=ami-4733f928
eu-west-1=ami-89f6dbef
eu-west-2=ami-052b3e61
sa-east-1=ami-c51573a9
us-east-1=ami-42ad7d54
us-east-2=ami-69a3860c
us-west-1=ami-1a1b457a
us-west-2=ami-2551d145
us-gov-west-1=ami-a846fcc9
```

An issue I ran into was that the latest 2016 natami from https://aws.amazon.com/amazon-linux-ami/
wouldn't run out of the box with our CF templates.

**Pull request closed without merging**